modules/social_features/social_post/src/PostAccessControlHandler.php

```diff
@@ -116,8 +116,8 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
                     $group_roles = $group_role_storage->loadByUserAndGroup($account, $group);
                     /** @var \Drupal\group\Entity\GroupRoleInterface $group_role */
                     foreach ($group_roles as $group_role) {
-                      if ($group_role->isOutsider() && $group_role->hasPermission($permission)) {
-                        return AccessResult::allowed()->cachePerUser()->addCacheableDependency($entity);
+                      if ($group_role->isOutsider()) {
```

Comment:
Will admins / SM still be able to check it out?

Reply:
I've checked and no - CM/SM, which are not group members don't have access to Group only posts after that.
So this probably needs to be fixed in a different way, @denysyukm.

Reply:
So I did some investigation and found that if we revoke "access posts in group" permission for outsiders in flexible_group then everything will work properly. And no needed to do changes in this PostAccessControllHandler.